test/fix.js

```diff
@@ -197,6 +197,20 @@ for (const [name, testFix] of Object.entries(testMethods)) {
         const { content } = await testFix(t, testdir)
         t.strictSame(content.scripts, { test: 'test-script' })
       })
+      t.test('no scripts', async t => {
+        const testdir = {
+          'package.json': pkg({ scripts: undefined }),
+        }
+        const { content } = await testFix(t, testdir)
+        t.notHas(content, 'scripts')
+      })
+      t.test('falsy scripts', async t => {
+        const testdir = {
+          'package.json': pkg({ scripts: 0 }),
+        }
+        const { content } = await testFix(t, testdir)
+        t.strictSame(content.scripts, undefined)
```

Comment:
This is what I'm talking about w/ the change in behavior.  Today npm would do this:

```sh
~/D/n/s/a $ npm pkg get scripts
0
~/D/n/s/a $ npm pkg fix
~/D/n/s/a $ npm pkg get scripts
{}
```

Currently consumers can reasonably assume that the `scripts` attribute in a registry manifest is an object. Even if empty.

Reply:
I might be wrong here but if package.json has `"scripts":0` and we run `npm pkg fix` it just removes the `scripts` from package.json currently in latest
the reason it `npm pkg get scripts` returns `{ }` is the design of get command if it doesn't find any value for that field.

Reply:
```
$ cat package.json
{
  "name": "a",
  "version": "1.0.0",
  "main": "index.js",
  "keywords": [],
  "author": "Gar <gar+npm@danger.computer>",
  "license": "ISC",
  "description": ""
}
```

Yep! Good catch.  I think I may have gotten that confused w/ the normalization done during publish.

```
    } else {
      changes?.push(`Removed invalid "scripts"`)
      delete data.scripts
    }
```

It's like this in `scriptpath` which is what `prepare` does (the task done during publish).